Tests/tests/test_TraceControl.cpp

```diff
@@ -0,0 +1,420 @@
+                                                    "\"module\":\".+\","
+                                                    "\"category\":\".+\","
+                                                    "\"state\":\"(disabled|enabled|tristated)\""
+                                                    "\\}")));
+    }
+
+    //Set Plugin_TraceControl:Information:enabled
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("set"), _T("{\"module\":\"Plugin_TraceControl\",\"category\":\"Information\",\"state\":\"enabled\"}"), response));
+
+    //Get status Plugin_TraceControl:Information:enabled
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("status"), _T("{\"module\":\"Plugin_TraceControl\",\"category\":\"Information\"}"), response));
+    params.FromString(response);
+    arr = params["settings"].Array();
+    traceObj = arr[0].Object();
+    traceObj.ToString(jsonString);
+    EXPECT_THAT(jsonString, ::testing::MatchesRegex(_T("\\{"
+                                                "\"module\":\"Plugin_TraceControl\","
+                                                "\"category\":\"Information\","
+                                                "\"state\":\"enabled\""
+                                                "\\}")));
+
+    //Set Plugin_TraceControl:All:disabled
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("set"), _T("{\"module\":\"Plugin_TraceControl\",\"state\":\"disabled\"}"), response));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("status"), _T("{\"module\":\"Plugin_TraceControl\"}"), response));
+    params.FromString(response);
+    arr = params["settings"].Array();
+    //Check all categories are set to disabled
+    for (unsigned int i = 0; i < arr.Length(); i++)
+    {
+        traceObj = arr[i].Object();
+        traceObj.ToString(jsonString);
+        EXPECT_THAT(jsonString, ::testing::MatchesRegex(_T("\\{"
+                                                    "\"module\":\"Plugin_TraceControl\","
+                                                    "\"category\":\".+\","
+                                                    "\"state\":\"disabled\""
+                                                    "\\}")));
+    }
+
+    //Set Plugin_TraceControl:All:enabled
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("set"), _T("{\"module\":\"Plugin_TraceControl\",\"state\":\"enabled\"}"), response));
+
+    //Log some trace data
+    TRACE(Trace::Information, (_T("Test1")));
+    TRACE(Trace::Information, (_T("Test2")));
+    TRACE(Trace::Information, (_T("Test3")));
+
+    //Access trace file from the system
+    Core::File file(volatilePath+"tracebuffer");
+    file.Open(true);
+    uint8_t buffer[1024];
+    uint32_t len = file.Read(buffer, sizeof(buffer));
+    std::string s(buffer, buffer+len);
+
+    //Verify the test data is present in the trace file
+    bool result = (s.find("Test1") != string::npos);
```

Comment:
please make regex instead

Reply:
The trace buffer is a circular buffer allocated in Thunder. The Trace() function writes data into that buffer and in this line [](https://github.com/rdkcentral/rdkservices/blob/sprint/22Q4/TraceControl/TraceOutput.h#L65), the data is retrieved from the buffer and outputs to console (printf). Since I cannot capture this console output, I am reading this trace buffer and matching for the test string.

Reply:
when unit testing need to find smallest unit and test its interface e.g. input/output. In your case unit is plugin and output is printf/syslog. The buffer file seems to be internal implementation of the unit . If owners rewrite their implementation to not use buffer file it would be complicated to fix your test. Please find a better solution that does not rely on the internal implementation. That could be wrap of syslog, for example

Reply:
I couldn't figure out on capturing data form syslog or printf to validate the trace output. Tried mock for TraceOutput, but it didn't help to override call to Output() function. I removed this part of the code.

Reply:
example with system():
https://github.com/rdkcentral/rdkservices/blob/e85f325faae4ab85cc963a2fe8d75292298adbab/Tests/tests/test_SystemServices.cpp#L487

Reply:
I reviewed it but it didn't work for TraceOutput.

Reply:
Added mock for syslog in the latest commit.